org.eclipse.lsp4e/src/org/eclipse/lsp4e/enablement/EnablementTester.java

```diff
@@ -69,7 +72,7 @@ public boolean evaluate(@Nullable URI uri) {
 		IResource resource = null;
 		try {
 			IDocument document = null;
-			resource = LSPEclipseUtils.findResourceFor(uri);
```

Comment:
I think there are several places where we map resources to URIs.  Should we not benefit by having the cache reused by all consumers of the method? If so, what about building the cache as an static map in `LSPEclipseUtils`?